x/epochs/types/genesis.go

```diff
@@ -74,20 +74,34 @@ func DefaultGenesis() *GenesisState {
 func (gs GenesisState) Validate() error {
 	epochIdentifiers := map[string]bool{}
 	for _, epoch := range gs.Epochs {
-		if epoch.Identifier == "" {
-			return errors.New("epoch identifier should NOT be empty")
+		if err := epoch.Validate(); err != nil {
+			return err
 		}
 		if epochIdentifiers[epoch.Identifier] {
 			return errors.New("epoch identifier should be unique")
 		}
-		if epoch.Duration == 0 {
-			return errors.New("epoch duration should NOT be 0")
-		}
-		// enforce EpochCountingStarted is false for all epochs
-		if epoch.EpochCountingStarted {
-			return errors.New("epoch counting should NOT be started at genesis")
-		}
 		epochIdentifiers[epoch.Identifier] = true
 	}
 	return nil
 }
+
+// Validate also validates epoch info.
+func (epoch EpochInfo) Validate() error {
+	if epoch.Identifier == "" {
+		return errors.New("epoch identifier should NOT be empty")
+	}
+	if epoch.Duration == 0 {
+		return errors.New("epoch duration should NOT be 0")
+	}
+	// enforce EpochCountingStarted is false for all epochs
+	if epoch.EpochCountingStarted {
```

Comment:
this should be genesis only validation

Reply:
True, so we still need to have one additional check at initGenesis.